components/Authentication/signup.jsx

```diff
@@ -60,46 +59,90 @@ const ShowText = styled.Text`
   const [email, setEmail] = React.useState('');
+  const [validEmail, setValidEmail] = React.useState(false);
   const [password, setPassword] = React.useState('');
+  const [validPassword, setValidPassword] = React.useState(false);
   const [hidePassword, setHidePassword] = React.useState(true);
+  const isFirstRunName = React.useRef(true);
+  const isFirstRunEmail = React.useRef(true);
+  const isFirstRunPassword = React.useRef(true);
+
+  // TODO: this regex isn't fully functional
+  function ValidateEmail(mail) {
+    // eslint-disable-next-line
+    const re = /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    if (re.test(mail)) {
+      return (true);
+    }
+    return (false);
+  }
+
+  React.useEffect(() => {
+    if (isFirstRunName.current) {
+      isFirstRunName.current = false;
+      return;
+    }
+    setValidName(!((name.length > 1)));
+  }, [name]);
+
+  React.useEffect(() => {
+    if (isFirstRunEmail.current) {
+      isFirstRunEmail.current = false;
+      return;
+    }
+    setValidEmail(!ValidateEmail(email));
+  }, [email]);
+
+  React.useEffect(() => {
+    if (isFirstRunPassword.current) {
+      isFirstRunPassword.current = false;
+      return;
+    }
+    setValidPassword(!(password.length > 5));
+  }, [password]);
 
   function signupPress() {
-    firebase.auth().createUserWithEmailAndPassword(email, password)
-      .catch((error) => {
-        const errorCode = error.code;
-        const errorMessage = error.message;
-        switch (errorCode) {
-          case 'auth/weak-password':
-            alert('Password too weak.');
-            break;
-
-          case 'auth/invalid-email':
-            alert('Invalid email.');
-            break;
-
-          case 'auth/email-already-in-use':
-            alert('That email is already being used.');
-            break;
-
-          default:
-            alert(`Error: ${errorMessage}`);
-        }
-      });
+    if (name.length > 1) {
+      firebase.auth().createUserWithEmailAndPassword(email, password)
+        .then((user) => {
+          user.user.updateProfile({
+            displayName: name,
+          }).then(() => {
+            navigation.navigate('Root');
+          }).catch((error) => {
+            console.log('Display name not set.');
+            console.log(error);
+          });
+        })
+        .catch((error) => {
+          const errorMessage = error.message;
+          // TODO: this alert looks odd for some reason,
+          // can't remember what an invalid password alert looks like
+          Alert.alert('Error:', errorMessage);
+        });
+    } else {
+      alert('Please enter your name.');
+    }
   }
 
   function loginPress() {
     navigation.navigate('Login');
   }
 
-  firebase.auth().onAuthStateChanged(() => {
-    if (firebase.auth().currentUser) {
-      navigation.navigate('Root');
-    }
-  });
```

Comment:
Also I think removing this makes it so we don't auto-login if we're already logged in

Reply:
onAuthStateChanged only triggers on sign in or sign out, so it was redundant because when the login sign in resolves it will do what this does already. The auto-login happens through App.js, it won't even send them to signup or login if the device is logged in.